### DISCUSSION TITLE
WebIFCLoaderPlugin tutorial not available

### DISCUSSION THREAD

**Discussion by Yusuke-Togao** [xeokit/xeokit-sdk#1667] [Q&A]
To see where the IFC file opens in the xeokit-sdk, the WebIFCLoaderPlugin tutorial (https://xeokit.notion.site/Viewing-an-IFC-Model-with- WebIFCLoaderPlugin-9a572b801af949bf87a21c88968bd251), I used the .html code as is.

However, when I check the operation, the IFC file does not open and the console log shows ‘Uncaught Parameter expected: WebIFC’.

I found that the WebIFC parameter was not specified enough in the WebIFCLoaderPlugin constructor, so I tried to specify it, but the WebIFCLoaderPluginConfiguration object has a parameter named WebIFC It does not exist.

Does the tutorial shown above still work today? Is it a problem with my development environment or implementation?

`<!doctype html>
<html>
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>xeokit Example</title>
		<style>
      body {
          margin: 0;
          width: 100%;
          height: 100%;
          user-select: none;
      }
			#myCanvas {
			    width: 100%;
			    height: 100%;
			    position: absolute;
			    background: lightblue;
			    background-image: linear-gradient(lightblue, white);
			}
		</style>
</head>
<body>
    <canvas id="myCanvas"></canvas>
</body>
<script id="source" type="module">

import {Viewer, WebIFCLoaderPlugin} from 
"https://cdn.jsdelivr.net/npm/@xeokit/xeokit-sdk/dist/xeokit-sdk.es.min.js";

const viewer = new Viewer({
    canvasId: "myCanvas",
    transparent: true
});

viewer.camera.eye = [-3.933, 2.855, 27.018];
viewer.camera.look = [4.400, 3.724, 8.899];
viewer.camera.up = [-0.018, 0.999, 0.039];

const webIFCLoader = new WebIFCLoaderPlugin(viewer, {
    // Path to web-ifc.wasm, which does the IFC parsing for us
    wasmPath: "https://cdn.jsdelivr.net/npm/@xeokit/xeokit-sdk/dist/"
});

const model = webIFCLoader.load({ // Returns an Entity that represents the model
    src: "Duplex.ifc",
    edges: true
});

</script>
</html>`

**Comment by MichalDybizbanskiCreoox** [comment]
Hi @Yusuke-Togao,

It looks like the tutorial page uses an outdated interface to demonstrate `WebIFCLoaderPlugin` use, sorry about that!

The `WebIFCLoaderPlugin` constructor relies on `WebIFC` and `IfcAPI` objects being supplied but those are not provided by the `xeokit-sdk` anymore (this changed at https://github.com/xeokit/xeokit-sdk/commit/10ce6cf8eb4ea1b72c2c7075c09ef6e56586b6a2).
Instead they can be loaded from another location and then passed to the `WebIFCLoaderPlugin` constructor.

One of examples that uses `WebIFCLoaderPlugin` to load its model is is https://xeokit.github.io/xeokit-sdk/examples/buildings/#web-ifc_vbo_Duplex.

At line 181 a `WebIFC` implementation is loaded:
https://github.com/xeokit/xeokit-sdk/blob/master/examples/buildings/web-ifc_vbo_Duplex.html#L181

At lines 206-215 this API is used to create and initialize an `IfcAPI` object, which is later passed to the `WebIFCLoaderPlugin` constructor:
https://github.com/xeokit/xeokit-sdk/blob/master/examples/buildings/web-ifc_vbo_Duplex.html#L206-L215

After that the `ifcLoader` object is ready to be used to load `.ifc` models. 

Hope that helps.
Best,
  Michał
